protocol-designer/src/pages/ProtocolOverview/ScrubberContainer.tsx

```diff
@@ -0,0 +1,172 @@
+import { useSelector } from 'react-redux'
+import flatMap from 'lodash/flatMap'
+import { ProtocolTimelineScrubber } from '@opentrons/components'
+import { OT2_ROBOT_TYPE } from '@opentrons/shared-data'
+import {
+  getInitialRobotState,
+  getRobotStateTimeline,
+  getRobotType,
+} from '../../file-data/selectors'
+import { uuid } from '../../utils'
+import {
+  getInitialDeckSetup,
+  getInvariantContext,
+} from '../../step-forms/selectors'
+import { getLabwareNicknamesById } from '../../ui/labware/selectors'
+import { selectors as ingredSelectors } from '../../labware-ingred/selectors'
+import { getLoadCommands } from '../../file-data/selectors/utils'
+import type {
+  AddressableAreaName,
+  CompletedProtocolAnalysis,
+  LabwareLocation,
+  Liquid,
+  LoadModuleRunTimeCommand,
+  LoadedLabware,
+  LoadedModule,
+  LoadedPipette,
+  RunTimeCommand,
+} from '@opentrons/shared-data'
+
+export function ScrubberContainer(): JSX.Element | null {
+  const robotType = useSelector(getRobotType)
+  const labwareNickNames = useSelector(getLabwareNicknamesById)
+  const robotStateTimeline = useSelector(getRobotStateTimeline)
+  const initialRobotState = useSelector(getInitialRobotState)
+  const ingredients = useSelector(ingredSelectors.getLiquidGroupsById)
+  const ingredientLocations = useSelector(ingredSelectors.getLiquidsByLabwareId)
+  const invariantContext = useSelector(getInvariantContext)
+  const initialDeckSetup = useSelector(getInitialDeckSetup)
+
+  if (robotType === OT2_ROBOT_TYPE) {
+    return null
+  }
+
+  const { pipetteEntities, labwareEntities, moduleEntities } = invariantContext
+  const {
+    pipettes,
+    modules,
+    labware,
+    additionalEquipmentOnDeck,
+  } = initialDeckSetup
+
+  const loadCommands = getLoadCommands(
+    initialRobotState,
+    pipetteEntities,
+    moduleEntities,
+    labwareEntities,
+    labwareNickNames,
+    ingredients,
+    ingredientLocations
+  )
+  const nonLoadCommands = flatMap(
+    robotStateTimeline.timeline,
+    timelineFrame => timelineFrame.commands
+  )
+  const runTimeCommands: RunTimeCommand[] = [
+    ...loadCommands,
+    ...nonLoadCommands,
+  ].map(command => {
+    let result
+    if (command.commandType === 'loadModule') {
+      const loadModuleResult: LoadModuleRunTimeCommand['result'] = {
+        moduleId: command.params.moduleId ?? '',
+      }
```

Comment:
maybe this is cleaner
```suggestion
      const loadModuleResult: LoadModuleRunTimeCommand['result'] | null = command.params.moduleId != null ? {
        moduleId: command.params.moduleId
      } : null
```